spec/controllers/cars_controller_spec.rb

```diff
@@ -0,0 +1,119 @@
+    it { expect(get: '/api/cars/1').to route_to(controller: 'cars', format: :json, action: 'show', id: '1') }
+    it { expect(get: '/api/cars/new').to_not route_to(controller: 'cars', action: 'new', format: :json) }
+    it { expect(patch: '/api/cars/1').to_not route_to(controller: 'cars', action: 'update', format: :json,id: '1') }
+    it { expect(put: '/api/cars/1').to_not route_to(controller: 'cars', action: 'update', format: :json, id: '1') }
+    it { expect(get: '/api/cars/1/edit').to_not route_to(controller: 'cars', action: 'edit', format: :json, id: '1') }
+  end
+
+
+  let(:car) { create(:car) }
+  let(:token) { JsonWebToken.encode(user_id: car.user.id) }
+
+  describe 'GET#index' do
+    let(:send_request) { get :index }
+    let(:expected_response) do
+      [car.present.cars_page_context].to_json
+    end
+
+    before do
+      request.headers['token'] = token
+      send_request
+    end
+
+    it { expect(response.content_type).to include('application/json') }
+    it { expect(response).to have_http_status(:success) }
+    it 'render JSON with list of car' do
+      expect(response.body).to eq(expected_response)
+    end
+  end
+
+  describe 'GET#show' do
+    let(:send_request) { get :show, params: { id: car.id } }
+    let(:expected_response) do
+      car.present.cars_page_context.to_json
+    end
+
+    before do
+      request.headers['token'] = token
+      send_request
+    end
+    it { expect(response.content_type).to include('application/json') }
+    it { expect(response).to have_http_status(:success) }
+    it 'return JSON with car info' do
+      expect(response.body).to eq(expected_response)
+    end
+  end
+
+  describe 'POST#create' do
+    let(:send_request) { post :create, params: { car: car_params } }
+    context 'when car' do
+      context 'is saved' do
+        let(:car_params) do
+          {
+            manufacturer: Faker::Vehicle.manufacture,
+            model: Faker::Vehicle.model,
+            year:  Faker::Number.number(digits: 4),
+            color:  Faker::Color.hex_color,
+            license_plat_number:  Faker::Number.number(digits: 4),
+            picture:  Rack::Test::UploadedFile.new('spec/support/assets/test-image.jpeg', 'image/jpeg')
+          }
+        end
+
+        before do
+          request.headers['token'] = token
+          send_request
+        end
+
+        it { expect(response.content_type).to include('application/json') }
+        it { expect(response).to have_http_status(:created) }
+      end
+
+      context 'is not saved' do
+        let(:error_message) { { picture: ["can't be blank"] } }
+        let(:expected_response) { service_responce.to_json }
+        let(:car_params) do
+          {
+            manufacturer: Faker::Vehicle.manufacture,
+            model: Faker::Vehicle.model,
+            year:  Faker::Number.number(digits: 4),
+            color:  Faker::Color.hex_color,
+            license_plat_number:  Faker::Number.number(digits: 4),
+          }
```

Comment:
This hash is almost similar with above one. You can define it once and reuse. For example:
```ruby
let(:hash) { {a: 1} }
context do
  let(:hash) { super().merge(b: 2) }
  
  p hash
  # => {a: 1, b: 2}
end

context do
  let(:hash) { super().merge(c: 3) }
  
  p hash
  # => {a: 1, c: 3}
end